performance/tests/Button.test.js

```diff
@@ -1,82 +1,139 @@
-			await page.tracing.start({path: filename, screenshots: false});
 			await page.waitForSelector('#button');
 			await page.focus('#button');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 
-			await page.tracing.stop();
+			const averageFPS = (FPSValues.reduce((a, b) => a + b, 0) / FPSValues.length) || 0;
+			TestResults.addResult({component: component, type: 'Frames Per Second Keypress', actualValue: averageFPS});
+		});
+	});
+
+	it('should have a good FID and CLS', async () => {
+		await page.evaluateOnNewDocument(FID);
+		await page.evaluateOnNewDocument(CLS);
+		await page.goto('http://localhost:8080/button');
+		await page.waitForSelector('#button');
+		await page.focus('#button');
+		await page.keyboard.down('Enter');
+		await page.waitForTimeout(200);
 
-			const actualFPS = FPS(filename);
-			TestResults.addResult({component: 'Button', type: 'Frames Per Second', actualValue: actualFPS});
+		let actualFirstInput = await page.evaluate(() => {
+			return window.fid;
+		});
 
-			const actualUpdateTime = Update(filename, 'Touchable');
-			TestResults.addResult({component: 'Button', type: 'Update', actualValue: actualUpdateTime});
+		let actualCLS = await page.evaluate(() => {
+			return window.cls;
 		});
+
+		TestResults.addResult({component: component, type: 'First Input Delay', actualValue: actualFirstInput});
+		TestResults.addResult({component: component, type: 'CLS', actualValue: actualCLS});
+
+		expect(actualFirstInput).toBeLessThan(maxFID);
+		expect(actualCLS).toBeLessThan(maxCLS);
 	});
 
-	it('should mount Button under threshold', async () => {
-		const filename = getFileName('Button');
+	it('should have a good DCL, FCP and LCP', async () => {
+		const filename = getFileName(component);
 
-		await page.tracing.start({path: filename, screenshots: false});
-		await page.goto('http://localhost:8080/button');
-		await page.waitForSelector('#button');
+		let contDCL = 0;
+		let contFCP = 0;
+		let contLCP = 0;
+		let avgDCL = 0;
+		let avgFCP = 0;
+		let avgLCP = 0;
+		for (let step = 0; step < stepNumber; step++) {
+			const page = await testMultiple.newPage();
+
+			await page.tracing.start({path: filename, screenshots: false});
+			await page.goto('http://localhost:8080/button');
+			await page.waitForSelector('#button');
+			await page.waitForTimeout(200);
+
+			await page.tracing.stop();
+
+			const actualDCL = await DCL(filename);
```

Comment:
In the LCP, FCP,  DCL function, 
it opens the same file. 
```
const DCL = (filename) => {
	const events = fs.readFileSync(filename, 'utf8');
	const result = JSON.parse(events);
	const baseEvent = result.traceEvents.filter(i => i.name === 'TracingStartedInBrowser')[0].ts;
...
```

I was talked about this part, and since this is fileIO, I'm guessing it's going to be a time-consuming factor.